tests/tools/calendar_test.py

```diff
@@ -115,3 +117,66 @@ async def test_calendar_tool_crashes_if_too_many_days(httpd: HTTPServer):
     assert False
   except ValueError as e:
     assert str(e) == "next_days must be at most 366"
+
+class MockHTTPRequestHandler(SimpleHTTPRequestHandler):
+    """Custom handler to serve different calendar files for testing."""
+    
+    def do_GET(self):
+        if "test-calendar-1.ics" in self.path:
+            ics_path = path.join(path.dirname(__file__), "..", "fixtures", "test-calendar-1.ics")
+        elif "test-calendar-2.ics" in self.path:
+            ics_path = path.join(path.dirname(__file__), "..", "fixtures", "test-calendar-2.ics")
+        else:
+            self.send_error(404, "File not found")
+            return
```

Comment:
Let's rely at the number of requests sent instead of the path to serve a different calendar. This allows us to better mimic what will happen in the real world.

```suggestion
        if self.request_number == 0:
            ics_path = path.join(path.dirname(__file__), "..", "fixtures", "test-calendar-1.ics")
        else:
            ics_path = path.join(path.dirname(__file__), "..", "fixtures", "test-calendar-2.ics")
```